mlir/lib/Dialect/ArmSME/Transforms/TileAllocation.cpp

```diff
@@ -6,7 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 //
-// This pass allocates SME tiles at the 'func.func' op level for ArmSME
+// This transform allocates SME tiles at the 'func.func' op level for ArmSME
 // operations. It does this using a 16-bit tile mask that has a bit for each
 // 128-bit element tile (ZA0.Q-ZA15.Q), the smallest ZA tile granule.
```

Comment:
Please, could we have more documentation in this file? For example, a high level overview of how tile allocation works (live range calculation, followed by transformation). With some info for every section.